### PR TITLE
Fix redirection in httpclient (fixes #3895)

### LINKED ISSUE #3895: https post request
Seems like I can't make a post request. Here's what I want to do:
Request in Python:
```python
import requests

data = {'text': 'hello world', 'expire': '1m', 'lang': 'text'}
print(requests.post("https://ghostbin.com/paste/new",data=data).text)
```
Request in httpie
```sh
http post ghostbin.com/paste/new text='hello world' expire=1d lang=text --form
```
Attempt at a post request in Nim
```nim
import httpclient

var data = newMultipartData()
data["text"] = "hello world"
data["expire"] = "1m"
data["lang"] = "text"
echo postContent(url="https://ghostbin.com/paste/new", multipart=data)
# error
# Traceback (most recent call last)
# post_to_ghostbin.nim(7)  post_to_ghostbin
# httpclient.nim(553)      postContent
# httpclient.nim(536)      post
# httpclient.nim(449)      request
# httpclient.nim(432)      request
# net.nim(559)             connect
# nativesockets.nim(217)   getAddrInfo
# os.nim(135)              raiseOSError
# Error: unhandled exception: nodename nor servname provided, or not known  [OSError]
# Error: execution of an external program failed: '/Users/rsinayev/post_to_ghostbin '

echo(post("http://ghostbin.com/paste/new", body="text=Hello%20World&expire=10s&lang=text"))

# error
# post_to_ghostbin.nim(7, 10) template/generic instantiation from here
# lib/system.nim(2299, 24) Error: type mismatch: got (StringTableRef)
# but expected one of:
# system.$(x: int)
# system.$(x: int64)
# system.$(x: uint64)
# system.$(x: float)
# system.$(x: bool)
# system.$(x: char)
# system.$(x: cstring)
# system.$(x: string)
# system.$(x: Enum)
# system.$(x: T)
# $.$(x: Response)
# system.$(x: seq[T])
# system.$(x: set[T])
var extraHeaders = "Content-Type: application/x-www-form-urlencoded"
var body = "text=Hello%20World&expire=10s&lang=text"
echo(post("http://ghostbin.com/paste/new", body=body, extraHeaders=extraHeaders))
# same error
request("https://ghostbin.com/paste/new", httpMethod=httpPost, body="text=Hello%20World&expire=10s&lang=text")
# same error
```
nslookup:
```sh
> nslookup ghostbin.com
Server:	208.67.222.222
Address:	208.67.222.222#53

Non-authoritative answer:
Name:	ghostbin.com
Address: 52.91.215.199
```
comiled like
```sh
nim c -r -d:ssl post_to_ghostbin.nim
```
Am I doing something wrong?

### PR DESCRIPTION
**Pull request by def-**
As reported in https://github.com/nim-lang/Nim/issues/3895

### CODE REVIEW

File: lib/pure/httpclient.nim [hunk 1]
```diff
@@ -455,14 +455,15 @@ proc redirection(status: string): bool =
     if status.startsWith(i):
       return true
 
-proc getNewLocation(lastUrl: string, headers: StringTableRef): string =
+proc getNewLocation(lastURL: string, headers: StringTableRef): string =
```
Comment (dom96) [inline lib/pure/httpclient.nim]: Pretty sure the convention is actually `lastUrl`.

File: lib/pure/httpclient.nim [hunk 2]
```diff
@@ -528,14 +529,15 @@ proc post*(url: string, extraHeaders = "", body = "",
 
   result = request(url, httpPOST, xh, xb, sslContext, timeout, userAgent,
                    proxy)
-  var lastUrl = ""
+  var lastURL = url
   for i in 1..maxRedirects:
     if result.status.redirection():
+      echo "lastURL: ", lastURL
```
Comment (dom96) [inline lib/pure/httpclient.nim]: Remove this please.

Comment (def-) [inline lib/pure/httpclient.nim]: Oops, forgot to push my updated commit.

File: lib/pure/httpclient.nim (hunk 1 above)
Comment (def-) [inline lib/pure/httpclient.nim]: Sure, I just chose lastURL because it was more common in the code already.

**Comment by dom96** [comment]
Huh. The ``echo`` was kept. Oh well, I'll just remove it manually.

**Pull request merged**